### PR TITLE
style: Dates Tab Shift Due Dates Success SnackBar Design Changes

### PR DESCRIPTION
**Pull request by farhan-arshad-dev**
## Description

- Dates Tab Shift Due Dates Success SnackBar Design Changes

Jira: [LEARNER-9868](https://2u-internal.atlassian.net/browse/LEARNER-9868)
Github issue: [[Android] Shift Dates (PLS) Design Changes](https://github.com/openedx/openedx-app-android/issues/240)

| Old | New |
| ------------- | ------------- |
| Screenshot_20240320_155845 | Screenshot_20240320_155406 |

### CODE REVIEW

File: core/src/main/java/org/openedx/core/UIMessage.kt [hunk 1]
```diff
@@ -8,5 +8,7 @@ sealed class UIMessage {
         val duration: SnackbarDuration = SnackbarDuration.Long,
     ) : UIMessage()
 
+    class DatesShiftedSnackBar : UIMessage()
```
Comment (volodymyr-chekyrta) [inline core/src/main/java/org/openedx/core/UIMessage.kt:11]: What if we relocate class DatesShiftedSnackBar: UIMessage() within the course module? 
It appears unnecessary to inform the core module about Dates functionality.

File: core/src/main/java/org/openedx/core/ui/ComposeCommon.kt [hunk 2]
```diff
@@ -386,6 +387,59 @@ fun HandleUIMessage(
     }
 }
 
+@Composable
+fun DatesShiftedSnackBar(
+    showAction: Boolean = false,
+    onViewDates: () -> Unit? = {},
+    onClose: () -> Unit? = {},
+) {
```
Comment (volodymyr-chekyrta) [inline core/src/main/java/org/openedx/core/ui/ComposeCommon.kt:395]: Can be moved as well to the CourseUI.kt

File: course/src/main/java/org/openedx/course/presentation/dates/CourseDatesFragment.kt [hunk 3]
```diff
@@ -258,6 +264,12 @@ internal fun CourseDatesScreen(
             )
         }
 
+        val snackState = remember { SnackbarHostState() }
+        if (uiMessage is UIMessage.DatesShiftedSnackBar) {
+            LaunchedEffect(uiMessage) {
+                snackState.showSnackbar(message = "Dates Shifted", duration = SnackbarDuration.Long)
```
Comment (volodymyr-chekyrta) [inline course/src/main/java/org/openedx/course/presentation/dates/CourseDatesFragment.kt:270]: Please move this text to resources

Comment (farhan-arshad-dev) [reply course/src/main/java/org/openedx/course/presentation/dates/CourseDatesFragment.kt]: moved.

File: course/src/main/java/org/openedx/course/presentation/outline/CourseOutlineFragment.kt [hunk 4]
```diff
@@ -321,6 +303,12 @@ internal fun CourseOutlineScreen(
             )
         }
 
+        val snackState = remember { SnackbarHostState() }
+        if (uiMessage is UIMessage.DatesShiftedSnackBar) {
+            LaunchedEffect(uiMessage) {
+                snackState.showSnackbar(message = "Dates Shifted", duration = SnackbarDuration.Long)
```
[the remark this answers is not in the document]
Comment (farhan-arshad-dev) [reply course/src/main/java/org/openedx/course/presentation/outline/CourseOutlineFragment.kt]: moved.

File: course/src/main/java/org/openedx/course/presentation/ui/CourseUI.kt [hunk 5]
```diff
@@ -1157,6 +1158,59 @@ fun CourseDatesBannerTablet(
...
+        modifier = Modifier.padding(16.dp),
+        backgroundColor = MaterialTheme.appColors.background
+    ) {
+        Column(modifier = Modifier.padding(4.dp)) {
+            Box {
+                Text(
+                    modifier = Modifier
+                        .fillMaxWidth()
+                        .align(Alignment.CenterStart),
+                    text = stringResource(id = org.openedx.core.R.string.core_dates_shift_dates_successfully_title),
```
Comment (k1rill) [inline course/src/main/java/org/openedx/course/presentation/ui/CourseUI.kt:1177]: please move org.openedx.core.R to import `import org.openedx.core.R as CoreR`

File: course/src/main/java/org/openedx/course/presentation/ui/CourseUI.kt [hunk 6]
```diff
@@ -1157,6 +1158,59 @@ fun CourseDatesBannerTablet(
...
+                        contentDescription = "close",
+                        tint = MaterialTheme.appColors.onBackground,
+                    )
+                }
+            }
+            Text(
+                modifier = Modifier
+                    .padding(top = 4.dp)
+                    .fillMaxWidth(),
+                text = stringResource(id = org.openedx.core.R.string.core_dates_shift_dates_successfully_msg),
```
Comment (k1rill) [inline course/src/main/java/org/openedx/course/presentation/ui/CourseUI.kt:1193]: please move org.openedx.core.R to import import org.openedx.core.R as CoreR

File: course/src/main/java/org/openedx/course/presentation/ui/CourseUI.kt [hunk 7]
```diff
@@ -1157,6 +1158,59 @@ fun CourseDatesBannerTablet(
...
+                color = MaterialTheme.appColors.textFieldText,
+                style = MaterialTheme.appTypography.titleSmall,
+            )
+            if (showAction) {
+                OpenEdXOutlinedButton(
+                    modifier = Modifier
+                        .padding(top = 16.dp)
+                        .fillMaxWidth(),
+                    text = stringResource(id = org.openedx.core.R.string.core_dates_view_all_dates),
```
Comment (k1rill) [inline course/src/main/java/org/openedx/course/presentation/ui/CourseUI.kt:1202]: please move org.openedx.core.R to import import org.openedx.core.R as CoreR

File: course/src/main/java/org/openedx/course/presentation/ui/CourseUI.kt (hunk 5 above)
Comment (farhan-arshad-dev) [reply course/src/main/java/org/openedx/course/presentation/ui/CourseUI.kt]: moved.

File: course/src/main/java/org/openedx/course/presentation/ui/CourseUI.kt (hunk 6 above)
Comment (farhan-arshad-dev) [reply course/src/main/java/org/openedx/course/presentation/ui/CourseUI.kt]: moved.

File: course/src/main/java/org/openedx/course/presentation/ui/CourseUI.kt (hunk 7 above)
Comment (farhan-arshad-dev) [reply course/src/main/java/org/openedx/course/presentation/ui/CourseUI.kt]: moved.

File: core/src/main/java/org/openedx/core/UIMessage.kt (hunk 1 above)
Comment (farhan-arshad-dev) [reply core/src/main/java/org/openedx/core/UIMessage.kt]: relocated.

**Review by k1rill — APPROVED** [review]

**Review by volodymyr-chekyrta — APPROVED** [review]

**Pull request merged**